src/System.Memory/tests/Span/CopyTo.cs

```diff
@@ -175,12 +176,64 @@ public static void CopyToShorterArray()
         {
             int[] src = { 1, 2, 3 };
             int[] dst = { 99, 100 };
-            
+
             AssertThrows<ArgumentException, int>(src, (_src) => _src.CopyTo(dst));
             int[] expected = { 99, 100 };
             Assert.Equal<int>(expected, dst);  // CopyTo() checks for sufficient space before doing any copying.
         }
 
 
+        // This test case tests the Span.CopyTo method for large buffers of size 4GB or more. In the fast path,
+        // the CopyTo method performs copy in chunks of size 4GB (uint.MaxValue) with final iteration copying
+        // the residual chunk of size (bufferSize % 4GB). The inputs sizes to this method, 4GB and 4GB+256B,
+        // test the two size selection paths in CoptyTo method - memory size that is multiple of 4GB or,
+        // a multiple of 4GB + some more size.
+        [Theory]
+        [OuterLoop]
+        [PlatformSpecific(TestPlatforms.Windows | TestPlatforms.OSX)]
+        [InlineData(4L * 1024L * 1024L * 1024L)]
+        [InlineData((4L * 1024L * 1024L * 1024L) + 256)]
+        public static void CopyToLargeSizeTest(long bufferSize)
+        {
+            int GuidCount = (int)(bufferSize / Unsafe.SizeOf<Guid>());
+
+            unsafe
+            {
+                bool allocatedFirst = AllocationHelper.TryAllocNative((IntPtr)bufferSize, out IntPtr memBlockFirst);
```

Comment:
Because you are doing 2 allocations, they really should be inside the try block. Otherwise, you risk the first one succeeded, the second one throws an exception, and you end up not releasing the first one.